### PR TITLE
Add start of Vitest infrastructure for Home webview

### PR DESCRIPTION
**Pull request by dotNomad**
This PR adds the beginnings of test infrastructure, in Vitest, for our Vue Home webview view

## Type of Change

- - [ ] Bug Fix 
- - [ ] New Feature 
- - [ ] Breaking Change 
- - [ ] Documentation 
- - [ ] Refactor 
- - [x] Tooling 

## Approach

The approach here was to identify some utilities that didn't require much work at all to begin testing, and grab a core piece of code - in this case the `HostConduitService` and get some more complicated tests in place.

This sets up how we can use the Home store, the HostConduitService, and sending some messages (using Mocks where needed)

### CODE REVIEW

File: extensions/vscode/webviews/homeView/src/vscode.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { WebviewApi } from "vscode-webview";
+
+let apiInstance: WebviewApi<unknown> | undefined = undefined;
+
+export const vscodeAPI = (): WebviewApi<unknown> => {
```
Comment (dotNomad) [inline extensions/vscode/webviews/homeView/src/vscode.ts:5]: This was a bit weird. I tried to get `vitest` to play nicely with `acquireVsCodeApi()`, but I just couldn't get anything to work. 

So instead I created this function / singleton to call it so it was easily mockable as a module import function.

File: extensions/vscode/webviews/homeView/src/HostConduitService.test.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { App, createApp } from "vue";
+
+import { HostToWebviewMessageType } from "../../../src/types/messages/hostToWebviewMessages";
+import { WebviewToHostMessageType } from "../../../src/types/messages/webviewToHostMessages";
+import { useHostConduitService } from "src/HostConduitService";
+import { useHomeStore } from "src/stores/home";
+import { vscodeAPI } from "src/vscode";
+import { WebviewApi } from "vscode-webview";
+
+export function withSetup<T>(composable: () => T): App {
```
Comment (dotNomad) [inline extensions/vscode/webviews/homeView/src/HostConduitService.test.ts:12]: We can probably break this out into another util tests file at some point, but this helper created a simple App that used the composable so `onMount()` and similar lifecycle functions worked for composables.

**Comment by jonkeane** [comment]
Totally unrelated to the code but is this /vi:.tεst/ or /vit.εst/ (testing vi or the most vit?)

**Review by sagerb — APPROVED** [review]
Looks like a good beginning. Thanks for pulling this together.

I'll probably have some questions once I go to add some unit tests, but for now, this looks good enough to merge.

**Pull request merged**